docs/src/main/sphinx/connector/hive.rst

```diff
@@ -984,6 +984,108 @@ Procedures
+  :widths: 20, 60, 20
+  :header-rows: 1
+
+  * - Property name
+    - Description
+    - Default
+  * - ``auto_purge``
+    - The skip trash when table or partition is deleted.
+    -
+  * - ``avro_schema_url``
+    - The ``URI`` pointing to Avro schema for the table.
+    -
+  * - ``bucket_count``
+    - The number of buckets to group data into. 
+      Must be used with ``bucketed_by``.
+    -
+  * - ``bucketed_by``
+    - The bucketing column for the storage table. 
+      Must be used with ``bucketed_count``.
+    -
+  * - ``bucketing_version``
+    - The bucketing version.
+    -
+  * - ``csv_escape``
+    - CSV escape character. Can only be used with a ``CSV`` format table.
+    -
+  * - ``csv_quote``
+    - The ``CSV`` quote character. Can only be used with a ``CSV`` format table.
+    -
+  * - ``csv_separator``
+    - The ``CSV`` separator character. Can only be used with a ``CSV`` format 
+      table.
+    -
+  * - ``external_location``
+    - The file system, URI location for an external Hive table on S3, Azure Blob 
+      Storage, etc.
+    -
+  * - ``format``
+    - The table file format. Valid values include ``ORC``, ``PARQUET``, ``AVRO``, 
+      ``RCBINARY``, ``RCTEXT``, ``SEQUENCEFILE``, ``JSON``, ``TEXTFILE``, and 
+      ``CSV``. The catalog property ``hive.storage-format`` sets the default 
+      value and can change it to a different default.
+    - ``ORC``
+  * - ``null_format``
+    - The serialization format for ``NULL`` value. Can only be used with 
+      ``TEXTFILE``, ``RCTEXT``, or ``SEQUENCEFILE`` format tables.
+    -
+  * - ``orc_bloom_filter_columns``
+    - The ``ORC`` bloom filter index columns. Can only be used with 
+      ``ORC`` format tables.
+    - 
+  * - ``orc_bloom_filter_fpp``
+    - The ``ORC`` bloom filter false positive probability. Can only be used with
+      ``ORC`` format tables.
+    - 
+  * - ``partitioned_by``
+    - The partitioning column for the storage table. The columns listed in the 
+      ``partitioned_by`` clause must be the last columns as defined in the DDL. 
+      Otherwise, you get an error from Trino.
+    - ``[]``
+  * - ``skip_footer_line_count``
+    - The number of footer lines. Can only be used with ``TEXTFILE`` or ``CSV`` 
+      format tables.
+    - 
+  * - ``skip_header_line_count``
+    - The number of header lines. Can only be used with ``TEXTFILE`` or ``CSV`` 
+      format tables.
+    -
+  * - ``sorted_by``
+    - The column to sort by to determine bucketing for row. Can only be used if 
+      ``bucketed_by`` and ``bucket_count`` are specified as well.
+    - ``[]``
+  * - ``textfile_field_separator``
+    - The ``TEXTFILE`` field separator character. Can only be used with 
+      ``TEXTFILE`` format tables.
+    - 
+  * - ``textfile_field_separator_escape``
+    - The ``TEXTFILE`` field separator escape character. Can only be used with 
+      ``TEXTFILE`` format tables.
+    -
+  * - ``transactional``
+    - The table is transactional.
```

Comment:
I believe setting this property to true and using ORC file format with a supported metastore allows Trino to perform transactional operations on the table (same as Hive ACID?). @losipiuk will know more.

Reply:
It maps to Hive's `transactional` table property.

If the table is created via Trino it can be only used for ORC-based tables. In such case, ORC ACID tables are being created.
If one tries to use it with a different file format Trino will reject `CREATE` query.

Note: If one creates an old-style insert_only table via Hive then Trino will also show `transactinal=true` for that table. 
@findepi it feels like it would make sense to change semantics here. It can be confusing. WDYT?

Reply:
I don't believe we can simplify without lying about what table properties are set in the metastore. Hive uses `transactional = true` for both types of tables.

Reply:
Table properties are for defining tables. `transactional=true` is for creating ORC ACID transactional tables (format must be ORC as well). 

optionally, if we want, we can document that SHOW CREATE TABLE shows `transactional=true` also for insert-only tables.

i don't see a lie here.

Reply:
So in summary:

- Setting transactional = true + ORC table format = ORC ACID tables
- transactional = true for insert-only tables too (but we cannot create such tables via Trino)

Reply:
@hashhar do you know what happens if you CT with transactional=true, but with format != ORC ?
do we reject? does HMS reject?

Reply:
The metastore rejects (i.e. buggy metastores might allow doing so?).

e.g. on HDP3 PT env started as `./testing/bin/ptl env up --environment multinode --config config-hdp3`

```
trino:default> create table transactional_parquet (a int) with (transactional=true, format='PARQUET');
Query 20220214_111431_00016_6veaq failed: The table must be stored using an ACID compliant format (such as ORC): default.transactional_parquet
io.trino.spi.TrinoException: The table must be stored using an ACID compliant format (such as ORC): default.transactional_parquet
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.createTable(ThriftHiveMetastore.java:1084)
	at io.trino.plugin.hive.metastore.thrift.BridgingHiveMetastore.createTable(BridgingHiveMetastore.java:204)
	at io.trino.plugin.hive.metastore.cache.CachingHiveMetastore.createTable(CachingHiveMetastore.java:474)
	at io.trino.plugin.hive.HiveMetastoreClosure.createTable(HiveMetastoreClosure.java:165)
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore$CreateTableOperation.run(SemiTransactionalHiveMetastore.java:3259)
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore$Committer.executeAddTableOperations(SemiTransactionalHiveMetastore.java:2135)
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.commitShared(SemiTransactionalHiveMetastore.java:1472)
	at io.trino.plugin.hive.metastore.SemiTransactionalHiveMetastore.commit(SemiTransactionalHiveMetastore.java:1211)
	at io.trino.plugin.hive.HiveMetadata.commit(HiveMetadata.java:3382)
	at io.trino.plugin.hive.HiveTransactionManager.lambda$commit$0(HiveTransactionManager.java:59)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.trino.plugin.hive.HiveTransactionManager.commit(HiveTransactionManager.java:57)
	at io.trino.plugin.hive.HiveConnector.commit(HiveConnector.java:207)
	at io.trino.transaction.InMemoryTransactionManager$TransactionMetadata$ConnectorTransactionMetadata.commit(InMemoryTransactionManager.java:623)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:131)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:74)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:82)
	at io.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:80)
	at io.trino.$gen.Trino_370_86_ge4153b0_dirty____20220214_111237_2.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.apache.hadoop.hive.metastore.api.MetaException: The table must be stored using an ACID compliant format (such as ORC): default.transactional_parquet
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_result$create_table_resultStandardScheme.read(ThriftHiveMetastore.java:54418)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_result$create_table_resultStandardScheme.read(ThriftHiveMetastore.java:54386)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$create_table_result.read(ThriftHiveMetastore.java:54312)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:86)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_create_table(ThriftHiveMetastore.java:1539)
	at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.create_table(ThriftHiveMetastore.java:1526)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient.createTable(ThriftHiveMetastoreClient.java:176)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.lambda$createTable$7(FailureAwareThriftMetastoreClient.java:132)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.runWithHandle(FailureAwareThriftMetastoreClient.java:474)
	at io.trino.plugin.hive.metastore.thrift.FailureAwareThriftMetastoreClient.createTable(FailureAwareThriftMetastoreClient.java:132)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.lambda$createTable$51(ThriftHiveMetastore.java:1072)
	at io.trino.plugin.hive.metastore.thrift.ThriftMetastoreApiStats.lambda$wrap$0(ThriftMetastoreApiStats.java:42)
	at io.trino.plugin.hive.util.RetryDriver.run(RetryDriver.java:130)
	at io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore.createTable(ThriftHiveMetastore.java:1070)
	... 22 more

trino:default> create table transactional_orc (a int) with (transactional=true, format='ORC');
CREATE TABLE
```